src/tezos.js

```diff
@@ -20,11 +20,17 @@ const metadataProvider = new MetadataProvider(
   ])
 );
 
-const Tezos = new TezosToolkit(rpcUrl);
+const buildTezos = (rpcUrl) => {
```

Comment:
I think this method should be memoized to avoid the same steps for each metadata query for, for example, mainnet tokens